core/src/com/unciv/logic/civilization/CivInfoStats.kt

```diff
@@ -32,17 +32,38 @@ class CivInfoStats(val civInfo: CivilizationInfo) {
             unitsToPayFor = unitsToPayFor.filterNot {
                 it.getTile().isCityCenter() && it.canGarrison()
             }
-
-
-        var numberOfUnitsToPayFor = max(0f, unitsToPayFor.count().toFloat() - freeUnits)
-
-        for (unique in civInfo.getMatchingUniques(UniqueType.UnitMaintenanceDiscount, StateForConditionals(civInfo))) {
-            val numberOfUnitsWithDiscount = min(
-                numberOfUnitsToPayFor,
-                unitsToPayFor.count { it.matchesFilter(unique.params[1]) }.toFloat()
-            )
-            numberOfUnitsToPayFor += numberOfUnitsWithDiscount * unique.params[0].toFloat() / 100f
+        // Handle unit maintenance discounts
+        // Have to capture global and per-unit
+        // Free Garrison already removed above from sequence
+        // Initialize maintenance cost per unit, default 1 aka 100%
+        // Note all discounts are in the form of -X%, such as -25 for 25% reduction
+        for (unit in unitsToPayFor){
```

Comment:
Foreach unit
   Maintenance=1
   For(unique in getMatchingUniques(..)) maintenance *= unique.params[]. toPercent ()